platform/darwin/src/MGLLocationManager.h

```diff
@@ -0,0 +1,47 @@
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MGLLocationManagerDelegate;
+
+/**
+ The `MGLLocationManager` protocol defines a set of methods that you
+ use to receive location-related events.
+ */
+@protocol MGLLocationManager <NSObject>
+
+@required
+
+@property (nonatomic, weak) id<MGLLocationManagerDelegate> delegate;
+@property (nonatomic) CLDeviceOrientation headingOrientation;
+@property (nonatomic, readonly) CLAuthorizationStatus authorizationStatus;
+
+- (void)startUpdatingLocation;
+- (void)startUpdatingHeading;
+
+- (void)stopUpdatingLocation;
+- (void)stopUpdatingHeading;
+
+- (void)requestAlwaysAuthorization;
+- (void)requestWhenInUseAuthorization;
+
+@end
+
+/**
+ The `MGLLocationManagerDelegate` protocol defines a set of methods that you
+ use to receive location updates from the associated location manager.
+ */
+@protocol MGLLocationManagerDelegate <NSObject>
+
+- (void)locationManager:(id<MGLLocationManager>)manager
+     didUpdateLocations:(NSArray<CLLocation *> *)locations;
+
+- (void)locationManager:(id<MGLLocationManager>)manager
+       didUpdateHeading:(CLHeading *)newHeading;
+
+@optional
+
+@end
```

Comment:
I would suggest removing the usage of all `CoreLocation` types (including `CLAuthorizationStatus `, etc. — so that you can remove the `CoreLocation` import above). This has the following reasons:

* Some of them are difficult to initialize. `CLHeading` has no public initializer so that I was forced to create a subclass where I was overriding the `trueHeading` property. This was hacky and not obvious.
* You can only reuqest the values you really need. Instead of `CLHeading` cou can ask for the `trueHeading:(double):trueHeading`. This makes the API easier to integrate and simulate.
* You are getting completely independent of the `CoreLocation` framework

Reply:
Hi, @simonseyer. Thank you for your feedback, and agree with you. When I was designing the API I consider removing completely the dependency on CoreLocation (provide our version of `CLLocation`, `CLLocationCoordinate2D`,`CLHeading`, etc). Although making that change will require a new semver, and migrating code. 

I will follow up with a ticket related to remove CoreLocation dependency.